### PR TITLE
[FLINK-30659][hive] Drop flink-sql-parser-hive module

### PR DESCRIPTION
**Pull request by luoyuxia**
## What is the purpose of the change
The pr is to drop `flink-sql-parser-hive` module as it's not used any more.

## Brief change log
- Delete the module `flink-sql-parser-hive`
- Adjust the logic related to hive in flink-table-planner
- Move the constants from `flink-sql-parser-hive` to `flink-connector-hive`

## Verifying this change
This change is already covered by existing tests

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / **not documented**)

### CODE REVIEW

File: flink-connectors/flink-connector-hive/src/test/resources/endpoint/select.q [hunk 1]
```diff
@@ -47,7 +47,7 @@ CREATE TABLE hive_types_table (
     f7 DECIMAL(20, 8),
     f8 STRING,
     f9 VARCHAR(20),
-    f10 TIMESTAMP(9),
+    f10 TIMESTAMP,
```
Comment (luoyuxia) [inline flink-connectors/flink-connector-hive/src/test/resources/endpoint/select.q:50]: Hive dialect doesn't support `TIMESTAMP(9)`. The origin code will fall to Flink default parser which  is not what we expect.

File: flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/connectors/hive/HiveDialectITCase.java [hunk 2]
```diff
@@ -372,7 +371,7 @@ public void testInsert() throws Exception {
         assertThat(results.toString())
                 .isEqualTo(
                         "[+I[1, 0, static], +I[1, 1, a], +I[2, 0, static], +I[2, 1, b], +I[3, 0, static], +I[3, 1, c]]");
-        tableEnv.executeSql("insert overwrite dest2 partition (p1,p2) select 1,x,y from src")
+        tableEnv.executeSql("insert overwrite table dest2 partition (p1,p2) select 1,x,y from src")
```
Comment (luoyuxia) [inline flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/connectors/hive/HiveDialectITCase.java:374]: Actually, `insert overwrite dest2 partition` is not supported in Hive dialect. We need to change it to `insert overwrite table dest2 partition`.
The origin code will fall back to the parser  implemented in  flink-sql-parser which is not what we expect.

File: flink-table/flink-sql-parser/src/main/java/org/apache/flink/sql/parser/validate/FlinkSqlConformance.java [hunk 3]
```diff
@@ -60,24 +54,12 @@ public boolean isHavingAlias() {
...
     @Override
     public boolean isSortByAlias() {
-        switch (this) {
-            case DEFAULT:
-            case HIVE:
-                return true;
-            default:
-                return false;
-        }
+        return false;
```
Comment (Aitozi) [inline flink-table/flink-sql-parser/src/main/java/org/apache/flink/sql/parser/validate/FlinkSqlConformance.java:62]: this should be true?

File: flink-table/flink-sql-parser/src/main/java/org/apache/flink/sql/parser/validate/FlinkSqlConformance.java [hunk 4]
```diff
@@ -60,24 +54,12 @@ public boolean isHavingAlias() {
...
     @Override
     public boolean isSortByOrdinal() {
-        switch (this) {
-            case DEFAULT:
-            case HIVE:
-                return true;
-            default:
-                return false;
-        }
+        return false;
```
Comment (Aitozi) [inline flink-table/flink-sql-parser/src/main/java/org/apache/flink/sql/parser/validate/FlinkSqlConformance.java:57]: this should be true?

**Review by Aitozi — COMMENTED** [review]
@luoyuxia thanks for this cleanup. Only left small comments. PTAL.

File: flink-table/flink-sql-parser/src/main/java/org/apache/flink/sql/parser/validate/FlinkSqlConformance.java (hunk 3 above)
Comment (luoyuxia) [reply flink-table/flink-sql-parser/src/main/java/org/apache/flink/sql/parser/validate/FlinkSqlConformance.java:62]: Good catch!

**Comment by luoyuxia** [comment]
@Aitozi Thanks for your reviewing. I have fixed your comments. PTAL again.

**Review by Aitozi — APPROVED** [review]
LGTM +1

**Comment by luoyuxia** [comment]
Merging....

**Pull request merged**